File: SDKs/googleapis.com/gmailpostmastertools/v1/python/src/sdk/models/shared/trafficstats.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import deliveryerror as shared_deliveryerror
from ..shared import ipreputation as shared_ipreputation
from ..shared import feedbackloop as shared_feedbackloop

class TrafficStatsDomainReputationEnum(str, Enum):
    REPUTATION_CATEGORY_UNSPECIFIED = "REPUTATION_CATEGORY_UNSPECIFIED"
    HIGH = "HIGH"
    MEDIUM = "MEDIUM"
    LOW = "LOW"
    BAD = "BAD"


@dataclass_json
@dataclasses.dataclass
class TrafficStats:
    r"""TrafficStats
    Email traffic statistics pertaining to a specific date.
    """
    
    delivery_errors: Optional[list[shared_deliveryerror.DeliveryError]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('deliveryErrors') }})
    dkim_success_ratio: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dkimSuccessRatio') }})
    dmarc_success_ratio: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dmarcSuccessRatio') }})
    domain_reputation: Optional[TrafficStatsDomainReputationEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('domainReputation') }})
    inbound_encryption_ratio: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('inboundEncryptionRatio') }})
    ip_reputations: Optional[list[shared_ipreputation.IPReputation]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ipReputations') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    outbound_encryption_ratio: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('outboundEncryptionRatio') }})
    spammy_feedback_loops: Optional[list[shared_feedbackloop.FeedbackLoop]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('spammyFeedbackLoops') }})
    spf_success_ratio: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('spfSuccessRatio') }})
    user_reported_spam_ratio: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userReportedSpamRatio') }})
    

File: SDKs/googleapis.com/gmailpostmastertools/v1beta1/java/lib/src/main/java/openapisdk/models/shared/TrafficStats.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * TrafficStats
 * Email traffic statistics pertaining to a specific date.
**/
public class TrafficStats {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("deliveryErrors")
    public DeliveryError[] deliveryErrors;
    public TrafficStats withDeliveryErrors(DeliveryError[] deliveryErrors) {
        this.deliveryErrors = deliveryErrors;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dkimSuccessRatio")
    public Double dkimSuccessRatio;
    public TrafficStats withDkimSuccessRatio(Double dkimSuccessRatio) {
        this.dkimSuccessRatio = dkimSuccessRatio;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dmarcSuccessRatio")
    public Double dmarcSuccessRatio;
    public TrafficStats withDmarcSuccessRatio(Double dmarcSuccessRatio) {
        this.dmarcSuccessRatio = dmarcSuccessRatio;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("domainReputation")
    public TrafficStatsDomainReputationEnum domainReputation;
    public TrafficStats withDomainReputation(TrafficStatsDomainReputationEnum domainReputation) {
        this.domainReputation = domainReputation;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("inboundEncryptionRatio")
    public Double inboundEncryptionRatio;
    public TrafficStats withInboundEncryptionRatio(Double inboundEncryptionRatio) {
        this.inboundEncryptionRatio = inboundEncryptionRatio;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ipReputations")
    public IpReputation[] ipReputations;
    public TrafficStats withIpReputations(IpReputation[] ipReputations) {
        this.ipReputations = ipReputations;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public TrafficStats withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("outboundEncryptionRatio")
    public Double outboundEncryptionRatio;
    public TrafficStats withOutboundEncryptionRatio(Double outboundEncryptionRatio) {
        this.outboundEncryptionRatio = outboundEncryptionRatio;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("spammyFeedbackLoops")
    public FeedbackLoop[] spammyFeedbackLoops;
    public TrafficStats withSpammyFeedbackLoops(FeedbackLoop[] spammyFeedbackLoops) {
        this.spammyFeedbackLoops = spammyFeedbackLoops;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("spfSuccessRatio")
    public Double spfSuccessRatio;
    public TrafficStats withSpfSuccessRatio(Double spfSuccessRatio) {
        this.spfSuccessRatio = spfSuccessRatio;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("userReportedSpamRatio")
    public Double userReportedSpamRatio;
    public TrafficStats withUserReportedSpamRatio(Double userReportedSpamRatio) {
        this.userReportedSpamRatio = userReportedSpamRatio;
        return this;
    }
}